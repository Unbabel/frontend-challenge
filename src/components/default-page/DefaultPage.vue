<script setup>
	defineProps({
		imageSrc: {
			type: String,
		},
		pageLabel: {
			type: String,
		},
		btnLabel: {
			type: String,
		},
		btnHandler: {
			type: Function,
		},
	});
</script>

<template>
	<div class="default-page-container flex-center">
		<img
			v-if="imageSrc"
			class="default-page-img"
			:src="imageSrc"
			alt="default-page-img"
		/>
		<span v-if="pageLabel" class="default-page-label">{{ pageLabel }}</span>
		<button
			v-if="btnLabel && btnHandler"
			class="default-page-btn"
			@click="btnHandler"
			:aria-label="btnLabel + ' Button'"
		>
			{{ btnLabel }}
		</button>
	</div>
</template>

<style>
	.default-page-container {
		width: 100%;
		height: 100%;
		justify-content: center;
		padding-top: 2rem;
		flex-direction: column;
		row-gap: 2rem;
	}

	.default-page-img {
		width: 35%;
	}

	.default-page-label {
		width: 80%;
		text-align: center;
		font-size: 1.3rem;
		font-weight: 600;
		color: var(--palette-dark);
	}

	.default-page-btn {
		font-size: 1rem;
		font-weight: 500;
		color: var(--palette-white);
		border: none;
		border-radius: 10px;
		background-color: var(--palette-purple);
		padding: 0.7rem;
		cursor: pointer;
		text-transform: uppercase;
	}

	@media (max-width: 768px) {
		.default-page-label,
		.default-page-btn {
			font-size: 0.8rem;
		}

		.default-page-img {
			width: 60%;
		}
	}
</style>
