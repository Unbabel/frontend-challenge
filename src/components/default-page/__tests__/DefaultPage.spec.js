import { describe, it, expect, beforeEach, vi } from "vitest";

import { mount } from "@vue/test-utils";

// Components
import DefaultPage from "../DefaultPage.vue";

const mockHandler = vi.fn();

describe("DefaultPage", () => {
	let wrapper;

	beforeEach(() => {
		// Arrange
		wrapper = mount(DefaultPage, {
			propsData: {
				imageSrc: "/path/to/image",
				pageLabel: "Default Page",
				btnLabel: "Click Me",
				btnHandler: mockHandler,
			},
		});
	});

	it("matches snapshot", () => {
		// Assert
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("renders properly", () => {
		// Act
		const div = wrapper.findAll("div");
		const image = wrapper.findAll("img");
		const span = wrapper.findAll("span");
		const button = wrapper.findAll("button");

		// Assert
		expect(div.exists()).toBe(true);
		expect(div).toHaveLength(1);
		expect(image.exists()).toBe(true);
		expect(image).toHaveLength(1);
		expect(span.exists()).toBe(true);
		expect(span).toHaveLength(1);
		expect(button.exists()).toBe(true);
		expect(button).toHaveLength(1);
	});

	[
		{ el: "div", clx: "default-page-container" },
		{ el: "div", clx: "flex-center" },
		{ el: "img", clx: "default-page-img" },
		{ el: "span", clx: "default-page-label" },
		{ el: "button", clx: "default-page-btn" },
	].forEach(({ el, clx }) =>
		it("has correct css classes", () => {
			// Act
			const element = wrapper.find(el);

			// Assert
			expect(element.classes(clx)).toBe(true);
		})
	);

	it("should call btnHandler on button click", () => {
		// Act
		wrapper.find("button").trigger("click");

		// Assert
		expect(mockHandler).toBeCalled();
		expect(mockHandler).toBeCalledTimes(1);
	});
});
