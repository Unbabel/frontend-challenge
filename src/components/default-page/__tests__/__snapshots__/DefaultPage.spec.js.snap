// Vitest Snapshot v1

exports[`DefaultPage > matches snapshot 1`] = `"<div class=\\"default-page-container flex-center\\"><img src=\\"/path/to/image\\" alt=\\"default-page-img\\" class=\\"default-page-img\\"><span class=\\"default-page-label\\">Default Page</span><button aria-label=\\"Click Me Button\\" class=\\"default-page-btn\\"> Click Me </button></div>"`;
